contracts/adapters/AaveMigrationExtension.sol

```diff
@@ -0,0 +1,578 @@
+     * @return liquidity The new liquidity amount as a result of the increase.
+     */
+    function increaseLiquidityPosition(
+        uint256 _underlyingSupplyLiquidityAmount,
+        uint256 _wrappedSetTokenSupplyLiquidityAmount,
+        uint256 _tokenId
+    )
+        external
+        onlyOperator
+        returns (uint128 liquidity)
+    {
+        liquidity = _increaseLiquidityPosition(
+            _underlyingSupplyLiquidityAmount,
+            _wrappedSetTokenSupplyLiquidityAmount,
+            _tokenId
+        );
+    }
+
+    /**
+     * @notice OPERATOR ONLY: Decreases and collects from a liquidity position in the Uniswap V3 pool.
+     * @param _tokenId The ID of the token for which liquidity is being decreased.
+     * @param _liquidity The amount of liquidity to decrease.
+     * @param _underlyingRedeemLiquidityMinAmount The minimum amount of the underlying token to be redeemed from the decreased liquidity.
+     * @param _wrappedSetTokenRedeemLiquidityMinAmount The minimum amount of the wrapped SetToken to be redeemed from the decreased liquidity.
+     */
+    function decreaseLiquidityPosition(
+        uint256 _tokenId,
+        uint128 _liquidity,
+        uint256 _underlyingRedeemLiquidityMinAmount,
+        uint256 _wrappedSetTokenRedeemLiquidityMinAmount
+    )
+        external
+        onlyOperator
+    {
+        _decreaseLiquidityPosition(
+            _tokenId,
+            _liquidity,
+            _underlyingRedeemLiquidityMinAmount,
+            _wrappedSetTokenRedeemLiquidityMinAmount
+        );
+    }
+
+
+    /**
+     * @notice OPERATOR ONLY: Migrates a SetToken's position from an unwrapped collateral asset to another SetToken 
+     * that consists solely of Aave's wrapped collateral asset
+     * @param _underlyingLoanAmount The amount of unwrapped collateral asset to be borrowed via flash loan.
+     * @param _underlyingSupplyLiquidityAmount The amount of unwrapped collateral asset to supply for liquidity.
+     * @param _wrappedSetTokenSupplyLiquidityAmount The amount of the wrapped SetToken to supply for liquidity.
+     * @param _tokenId The Uniswap V3 position token ID related to the Extension's liquidity position.
+     * @param _exchangeName The name of the exchange to perform the trade.
+     * @param _underlyingTradeUnits The amount of the unwrapped collateral asset (in SetToken units) being sent to the exchange.
+     * @param _wrappedSetTokenTradeUnits The amount of the wrapped SetToken (in SetToken units) being received from the exchange.
+     * @param _exchangeData Additional data required for executing the trade on the specified exchange.
+     * @param _underlyingRedeemLiquidityMinAmount The minimum amount of unwrapped collateral asset to redeem from liquidity decrease.
+     * @param _wrappedSetTokenRedeemLiquidityMinAmount The minimum amount of the wrapped SetToken to redeem from liquidity decrease.
+     */
+    function migrate(
+        uint256 _underlyingLoanAmount,
```

Comment:
UnderlyingLoanAmount could probably be calculated in contract. (As well as some of the other parameters).
Not sure though if it is worth the extra gas costs. 
What do you think ?

Reply:
Omitting to keep consistency between independent amount parameters for (liquidity, trade, flashloan). 

Only deduced amounts are related to issuance / redemption.